### DISCUSSION TITLE
Getting url on first render

### DISCUSSION THREAD

**Discussion by D-Thrane** [vercel/next.js#34236] [Help]
I'm trying to get the url on first render. I have a json file with allot of url's from an old website that redirects to what the url in on the new website. So if first render url is a.mywebsite.com/hello i need to check if that url is in my redirect.json file. If a match it should redirect to fx. mywebsite.com/a/hello.

I have no luck getting the first render url in my _documents.tsx or slug.tsx file :(

**Comment by wwwdepot** [comment]
Use middleware ([example](https://github.com/vercel/examples/tree/main/edge-functions/redirects-upstash))

**Reply by D-Thrane** [reply]
It also only looks at pathname. That does not work as I have to look at a.mywebsite.com/hello
